File: Data Structures & Algorithms/target-sum/submission-4.java
class Solution {
    public int findTargetSumWays(int[] nums, int target) {
        int n = nums.length;
        return dfs(n-1, 0, nums, target);
    }
    private int dfs(int i, int j, int[] nums, int target) {
        if (i < 0) {
            return j == target ? 1 : 0;
        }

        return dfs(i-1, j + nums[i], nums, target) + dfs(i - 1, j - nums[i], nums, target);
    }

}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        ans = 0
        numSet = set(nums)

        for num in numSet:
            below = num - 1
            if below not in numSet:
                current = 1
                while (num+1) in numSet:
                    current += 1
                    num += 1
                ans = max(ans, current)
        return ans        